hollow/src/main/java/com/netflix/hollow/api/producer/validation/ValidatorStatus.java

```diff
@@ -0,0 +1,147 @@
+/*
+ *
+ *  Copyright 2017 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.producer.validation;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Generated;
+
+import com.netflix.hollow.api.producer.HollowProducerListener.Status;
+/**
+ * *************************************
+ * NOTE: Beta API subject to change.   *
+ * *************************************
+ * 
+ * @author lkanchanapalli {@literal<lavanya65@yahoo.com>}
+ * 
+ * ValidationStatus has one instance per each validator that ran. 
+ * For now ValidationStatus builds these and sets toString on validator as the message (where validators provide any details).
+ * In next iteration this might be directly returned by validators.
+ */
+class ValidatorStatus {
```

Comment:
Wonder whether we can improve the names between ValidationStatus and ValidatorStatus.  They were so close that I thought they were duplicate classes.

Reply:
Yes I felt the same. Suggestions are welcome.

Reply:
Made some changes. More changes to come to make interaction between HollowProducer, individual validator status and overall validation status better.